packages/babel-preset-expo/src/resolve-package.ts

```diff
@@ -0,0 +1,44 @@
+import path from 'path';
+import resolveFrom from 'resolve-from';
+
+/**
+ * Resolve and import the @react-native/babel-preset package.
+ */
+export function requireUpstreamBabelPreset() {
+  const reactNativePackageJsonPath = require.resolve('react-native/package.json');
+  const reactNativePath = path.dirname(reactNativePackageJsonPath);
+  const babelPresetPath = resolveProjectTransitiveDependency(
+    reactNativePath,
+    '@react-native/community-cli-plugin',
+    '@react-native/metro-babel-transformer',
+    '@react-native/babel-preset'
+  );
+  if (!babelPresetPath) {
+    throw new Error('Unable to resolve the @react-native/babel-preset package.');
+  }
+  return require(babelPresetPath);
+}
+
+/**
+ * Resolve the path to a transitive dependency from the project.
+ */
+export function resolveProjectTransitiveDependency(
```

Comment:
This seems a little expensive for a babel plugin.

Reply:
makes sense! let me test more and share the profiling data about the resolving change.